src/SlackConnector/SlackConnection.cs

```diff
@@ -174,17 +202,35 @@ private void RaiseOnDisconnect()
         public event MessageReceivedEventHandler OnMessageReceived;
         private async Task RaiseMessageReceived(SlackMessage message)
         {
-            if (OnMessageReceived != null)
+            var e = OnMessageReceived;
```

Comment:
I don't understand why you have done this + why is the variable name is just `e`?